apps/api/src/database/repositories/competition-repository.ts

```diff
@@ -1758,39 +1759,49 @@ export const findActiveCompetitionsPastEndDate = createTimedRepositoryFunction(
 /**
  * Get portfolio timeline for agents in a competition
  * @param competitionId Competition ID
+ * @param bucket Time bucket interval in minutes (default: 30)
  * @returns Array of portfolio timelines per agent
  */
-async function getAgentPortfolioTimelineImpl(competitionId: string) {
+async function getAgentPortfolioTimelineImpl(
+  competitionId: string,
+  bucket: number = 30,
+) {
   try {
     const result = await dbRead.execute<{
-      date: string;
+      timestamp: string;
       agent_id: string;
       agent_name: string;
       competition_id: string;
       total_value: number;
     }>(sql`
-      SELECT DATE(timestamp) AS date, agent_id, name AS agent_name, competition_id, total_value FROM (
+      SELECT 
+        timestamp, 
+        agent_id, 
+        name AS agent_name, 
+        competition_id, 
+        total_value 
+      FROM (
         SELECT 
           ROW_NUMBER() OVER (
-            PARTITION BY agent_id, competition_id, DATE(timestamp)
+            PARTITION BY agent_id, competition_id,FLOOR(EXTRACT(EPOCH FROM (timestamp - competitions.start_date)) / 60 / ${bucket})
```

Comment:
We partition the table by **agent id**, **competition id**, and **buckets of ${bucket} minutes starting from competition start date**. Then we grab the last snapshot within that bucket. So, if we want to return 1 snapshot per day of competition we pass 1440, if we want 2, 720, ... default is 30.

cc @dtbuchholz